### ISSUE TITLE
Cover other Invenio RDM repos in Zenodo translator

### ISSUE THREAD

**Issue by marcschulder** [zotero/translators#3057]
There are a number of research data repositories that, like Zenodo, use Invenio RDM as their code base. Would it be possible to add them to the Zenodo translator or add a related translator? The former would be preferable with regard to code duplication, but I don't know how much the custom surface wrappers of the individual repositories or differences in how up-to-date their Invenio RDM version is might complicate things.

Examples of relevant repos are https://www.fdr.uni-hamburg.de/ (visually very close to Zenodo) and https://researchdata.tuwien.at/ (more heavily customised interface). I'd be happy to try and compile a longer list of eligible repos if that helps.

Tagging translator creators: @adam3smith @zuphilip

**Comment by dstillman** [comment]
Please don't tag people for general requests.

**Comment by adam3smith** [comment]
Yes, a full list would be great. Invenio URLs aren't distinctive enough to recognize based on regex. I know Caltech & NYU use Invenio here in the US

**Comment by marcschulder** [comment]
So this turned out to be rather more trickt than I had anticipated. I started out by looking at the [list of InvenioRDM contributors](https://inveniosoftware.org/products/rdm/), hoping that most would be hosting a repository. This was true in part, though several partners were service providers (I checked their customer lists where feasible) and in most cases the partner website hide its repo(s) pretty well. I eventually found a variety of repositories, though it was very difficult for me to determine whether they were actually running on InvenioRDM, some pre-RDM-rebrand Invenio version or some other software. In the end I mostly had to go by visual comparison. I also checked [the Invenio showcase](https://inveniosoftware.org/showcase/), but again ran into document repos that must evidently be running pre-DRM Invenio software.
Here are my likeliest candidates:

Strong match to Zotero UI:
- https://www.fdr.uni-hamburg.de/
- https://rodare.hzdr.de/
- https://aperta.ulakbim.gov.tr
- https://www.openaccessrepository.it/
- https://eic-zenodo.sdcc.bnl.gov/


Differing but similar UI:
- https://data.caltech.edu/
- https://repository.tugraz.at/
- https://researchdata.tuwien.at/
- https://ultraviolet.library.nyu.edu/
- https://adc.ei-basel.hasdai.org/
- https://vecnet.med.hasdai.org/
- https://miami.uni-muenster.de/
- https://fdat.uni-tuebingen.de/
- https://museumbollandianum.be/


Fairly different UI, but (probably) related:
- https://opendata.cern.ch/
- https://publications.jrc.ec.europa.eu/
- https://arch.library.northwestern.edu/
- https://harman.ulakbim.gov.tr/
- https://rogue-scholar.org/
- https://arodes.hes-so.ch/
- https://www.hepdata.net/
- https://b2share.eudat.eu/
- https://invenio.itam.cas.cz/
- https://phaidra.univie.ac.at/

Candidates that already have a custom translator:
- https://inspirehep.net/
- https://nyu.databrary.org

Special cases:
- [CERN CDS](https://cds.cern.ch/) is apparently also planning to switch to InvenioRDM, [according to their blog post](https://cds-blog.web.cern.ch/2023/01/the-new-cds-based-on-inveniordm/).
- Currently offline: https://invenio.ids-mannheim.de/
- _ data futures_ have a catalogue of [21 repositories](https://www.hasdai.org/search?page=1&size=10&q=&mode=sites). Two of these are in the above list, the other 19 had a very different specialised interface that focussed on showing individual image uploads and their metadata. 

Hope this helps. Let me know if I can do anything else.

**Comment by adam3smith** [comment]
Thanks! The first group will be easy. The second group requires fairly substantial additions to the code since nothing seems to work quite the same (e.g. it's records, not record, the metadata formats get exported rather than displayed inline) so while doable, that'd take a good chunk of work (PRs welcome). 

I don't see how the third group even makes sense within the same translator. The ones I looked at don't export to the same formats, so there's no overlap at all.

**Comment by aborel** [comment]
As far as I understand, Zenodo wasn't really using InvenioRDM when this issue was opened. It officially migrated to InvenioRDM in October 2023, and the Zenodo translator is now updated accordingly. Therefore, the easier targets should be group 2.

Servers in group 1 are probably using some variant of the weird Invenio version on which Zenodo was based. I assume that it would make sense for them to migrate to the actual InvenioRDM at some point, but obviously this is a non-trivial change and we cannot tell when they will be in a position to do it. Perhaps the previous Zenodo translator can be resurrected to support these repositories.

In group 3, I can say that https://arodes.hes-so.ch/ isn't using InvenioRDM. Like our own insitutional repository https://infoscience.epfl.ch/ , the platform is a variant of the old Invenio 1.2 (i.e. https://github.com/inveniosoftware/invenio instead of https://github.com/inveniosoftware/invenio-app-rdm)

**Comment by adam3smith** [comment]
Thanks for the context. @aborel would you be up for extending your Zenodo/Invenio translator to capture group 3 above?

**Comment by aborel** [comment]
@adam3smith Do you mean group 2?

**Comment by adam3smith** [comment]
yes, sorry

**Comment by aborel** [comment]
OK then :-)
And thanks to @marcschulder for listing the candidates!

**Comment by aborel** [comment]
I think this issue can be closed.
